src/slurm_plugin/instance_manager.py

```diff
@@ -1045,6 +1055,11 @@ def _launch_instances(
                                 print_with_count(batch_nodes),
                                 e,
                             )
+                            update_failed_nodes_parameters = {"nodeset": set(batch_nodes)}
+                            if isinstance(e, ClientError):
+                                update_failed_nodes_parameters["error_code"] = e.response.get("Error", {}).get("Code")
```

Comment:
Will there be any downstream impact of allowing any ClientError to be used as an `error_code`?
E.g. Event publishing etc.

Reply:
This is what is done in current behavior, that we want to keep consistent, see https://github.com/aws/aws-parallelcluster-node/blob/v3.6.1/src/slurm_plugin/instance_manager.py#L134